src/zcash/circuit/note.tcc

```diff
@@ -96,11 +96,6 @@ public:
         a_sk->generate_r1cs_constraints();
         rho->generate_r1cs_constraints();
 
-        // TODO: These constraints may not be necessary if SHA256
-        // already boolean constrains its outputs.
-        a_pk->generate_r1cs_constraints();
-        commitment->generate_r1cs_constraints();
-
```

Comment:
In those two's cases, since they are inputs, they are already constrained by the unpacker. I wouldn't mind going through and documenting that kind of stuff, but I wish a type system would do the talking.